arviz/plots/backends/matplotlib/ppplot.py

```diff
@@ -0,0 +1,102 @@
+"""Matplotib Posterior predictive plot."""
+import matplotlib.pyplot as plt
+import matplotlib
+import numpy as np
+
+from . import backend_show
+from ...kdeplot import plot_kde
+from ...plot_utils import _create_axes_grid
+
+colors = {
+    "prior": "red",
+    "posterior": "blue",
+    "observed": "yellow",
+}
+
+
+def plot_pp(
+    ax,
+    length_plotters,
+    rows,
+    cols,
+    figsize,
+    pp_plotters,
+    linewidth,
+    legend,
+    groups,
+    fill_kwargs,
+    plot_kwargs,
+    backend_kwargs,
+    show,
+):
+    """Matplotlib pp plot."""
+    if ax is None:
+        fig, ax = _create_axes_grid(
+            rows * cols, rows, cols, figsize=figsize, backend_kwargs=backend_kwargs
+        )
+        axes = []
+        for i in range(cols, rows * cols, 2 * cols):
+            axes += list(ax[i - cols : i])
+            gs = ax[i].get_gridspec()
+            for j in range(i, i + cols):
+                ax[j].remove()
+            axbig = fig.add_subplot(gs[i // cols, :])
+            axes.append(axbig)
```

Comment:
It feels cleaner to not use `_create_axes_grid` and use completely custom code with gridspec. idea in pseudocode:

```
fig = plt.figure(constrained_layout=rcParams[...])
gs = fig.add_gridspec(ncols=ngroups, nrows=nvars*2)
for i in range(nvars):
    for j in range(ngroups):
        ax_singles = [fig.add_subplot(gs[2*i, j])]
    ax_combined = fig.add_subplot(gs[2*i+1, :])
# combine axes into array, create array in a whole different way
# another option would be creating an array of shape (nvars, ngroup+1)
# so ax[i, :-1] = ax_singles and ax[i, -1] = ax_combined
```

The structure of the axes array has to be defined and explicited to create here, check shape and add to docstring. I think using a flattened array will be confusing.